tests/builder_test.cpp

```diff
@@ -0,0 +1,110 @@
+#include "monster_test_generated.h"
```

Comment:
just stick all this code in the header for simplicity? if not, there may be other build files that need updating, e.g. `BUILD`, Android etc.